src/Confluent.Kafka/Producer.cs

```diff
@@ -52,11 +52,15 @@ public class Producer : IDisposable
         private Task StartPollTask(CancellationToken ct)
             => Task.Factory.StartNew(() =>
                 {
-                    while (true)
+                    try
                     {
-                        ct.ThrowIfCancellationRequested();
-                        this.kafkaHandle.Poll((IntPtr)POLL_TIMEOUT_MS);
+                        while (true)
```

Comment:
Why not just this?:
```
  while (!ct.IsCancellationRequested) {
     this.kafkaHandle.Poll((IntPtr)POLL_TIMEOUT_MS);
  }
```

Reply:
see comments in #413

Reply:
And the reason for the AggregationException is still unknown?

Reply:
yes, I haven't looked into it again. I don't believe it is a symptom of something bigger.